packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js

```diff
@@ -193,30 +223,42 @@ const withInfiniteScroll = (
           return {
             isListFinished: true,
           };
+        })
+        .catch(err => {
+          if (onDataFetchFailed) {
+            onDataFetchFailed(err);
+          }
+          throw err;
         });
     }
 
     handleIntersection = (entries: Array<IntersectionObserverEntry>) => {
       const { onScroll } = this.props;
       const entry = entries[0];
       if (entry.isIntersecting) {
-        if (this.sentinel) {
-          this.observer.unobserve(this.sentinel);
-        }
+        this.unobserveSentinel();
         if (onScroll) {
           onScroll({ currentIndex: this.state.index });
         }
-        return this.fetchItems().then(newState => {
-          this.setState(newState);
-        });
+        return this.fetchItems()
+          .then(newState => {
+            this.setState(newState);
+          })
+          .catch(() => {
+            this.handleFetchItemsFailed();
```

Comment:
```suggestion
        .catch(this.handleFetchItemsFailed);
```
Bit shorter and has the same effect.